pkg/tiller/release_server.go

```diff
@@ -59,6 +59,8 @@ var (
 	errMissingRelease = errors.New("no release provided")
 	// errInvalidRevision indicates that an invalid release revision number was provided.
 	errInvalidRevision = errors.New("invalid release revision")
+	//errInvalidName indicates that an invalid release name was provided
+	errInvalidName = errors.New("invalid release name")
```

Comment:
any way we can return a more informative error as to why the release name is invalid? Something like "invalid release name. Must match regex \<insert ValidName regex here\>"